src/btree/bt_read.c

```diff
@@ -428,21 +516,21 @@ __las_page_instantiate(WT_SESSION_IMPL *session, WT_REF *ref)
      * instantiated.
      */
     for (i = 0, las_preparep = las_prepares->mem; i < las_prepare_cnt; i++, las_preparep++) {
-        cursor->set_key(
-          cursor, las_btree_id, &las_preparep->key, las_preparep->timestamp, las_preparep->txnid);
-        WT_ERR(cursor->remove(cursor));
+        las_cursor->set_key(las_cursor, las_btree_id, &las_preparep->key, las_preparep->timestamp,
+          las_preparep->txnid);
+        WT_ERR(las_cursor->remove(las_cursor));
     }
 
     __wt_buf_free(session, &page_las->max_las_key);
     __wt_buf_free(session, &page_las->min_las_key);
 
 err:
-    if (listp == NULL)
-        listp = list;
-    while (mod_counter > 0)
-        __wt_free_update_list(session, listp[--mod_counter]);
-    if (allocated_bytes != 0)
-        __wt_free(session, listp);
+    __wt_free_update_list(session, mod_upd);
+    while (modifies.size > 0) {
+        __wt_modify_vector_pop(&modifies, &mod_upd);
+        __wt_free_update_list(session, mod_upd);
+    }
+    __wt_modify_vector_free(&modifies);
```

Comment:
Would it be better for `__wt_modify_vector_free()` to free any remaining individual `WT_UPDATE` items as well?

Reply:
I considered this but there are some usages that store pointers to items in the update list. In that case, it's not ours to free.